nncf/data/generators.py

```diff
@@ -0,0 +1,85 @@
+# Copyright (c) 2024 Intel Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#      http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import List, TypeVar
+
+import nncf
+
+BASE_VOCAB_SIZE = 12000
+
+TModel = TypeVar("TModel")
+TTokenizer = TypeVar("TTokenizer")
+
+
+def generate_text_data(
+    model: TModel, tokenizer: TTokenizer, seq_len: int = 32, dataset_size: int = 128, unique_tokens_lower_limit: int = 5
+) -> List[str]:
+    """
+    Generates text dataset based on the model output.
+
+    :param model: Model instance.
+    :param tokenizer: Tokenizer instance.
+    :param seq_len: Sequence length for generation.
+    :param dataset_size: Size of the data.
+    :return: List of the text data ready to use.
+    """
+
+    try:
+        import torch
+    except ImportError:
+        raise nncf.ModuleNotFoundError("torch is required in order to generate text data: `pip install torch`.")
+
+    try:
+        from transformers import PreTrainedModel
+        from transformers import PreTrainedTokenizerBase
+        from transformers.utils import logging
+
+        logging.set_verbosity_error()
+    except ImportError:
+        raise nncf.ModuleNotFoundError(
+            "transformers is required in order to generate text data: `pip install transformers`."
+        )
+
+    if not isinstance(model, PreTrainedModel.__bases__):
+        raise nncf.ValidationError("Model should be instance of the `transformers.PreTrainedModel`.")
+
+    if not isinstance(tokenizer, PreTrainedTokenizerBase.__bases__):
+        raise nncf.ValidationError("tokenizer should be instance of the `transformers.PreTrainedTokenizerBase`.")
+
+    generated_data = []
+
+    vocab_size_names = ["padded_vocab_size", "vocab_size"]
+    vocab_size = BASE_VOCAB_SIZE
+    for vocab_size_name in vocab_size_names:
+        if hasattr(model.config, vocab_size_name):
+            vocab_size = getattr(model.config, vocab_size_name)
+
+    step_num = max(1, vocab_size // dataset_size)
+    ids_counter = 0
+
+    while len(generated_data) < dataset_size:
+        # Creating the input for pre-generate step
+        input_ids = torch.tensor([[ids_counter % vocab_size]])
```

Comment:
This function will be used only for torch modules or for OV too?
May be will useful to add option to return numpy.array to avoid installation torch? (add argument pt_tensor)

Reply:
OV can pass torch tensor as the input too.
It is not possible to avoid torch installation due to `transformers.PreTrainedModel` being the input. 
See the source for this class - https://github.com/huggingface/transformers/blob/v4.44.2/src/transformers/modeling_utils.py#L1297, which contains `(nn.Module, ModuleUtilsMixin, GenerationMixin, PushToHubMixin, PeftAdapterMixin)` as the base classes.